tests/framework/extensions/clusters/clusters.go

```diff
@@ -194,6 +194,55 @@ func NewRKE2ClusterConfig(clusterName, namespace, cni, cloudCredentialSecretName
 	return v1Cluster
 }
 
+// NewK3SClusterConfig is a constructor for a apisV1.Cluster object, to be used by the rancher.Client.Provisioning client.
+func NewK3SClusterConfig(clusterName, namespace, cloudCredentialSecretName, kubernetesVersion string, machinePools []provisioning.RKEMachinePool) *provisioning.Cluster {
```

Comment:
Is it necessary to have a whole other function to instantiate a K3SClusterConfig? It looks very similar to the RKE2. Why not have a shared function that does this that is private, then have two separate functions for RKE2 and K3S.

Reply:
K3s doesn't use cni, so there are some fields that need to be omitted and I figured it's easier to just have a new function.

Reply:
I think we can have on shared function that the public K3/RKE2 cluster functions would use to setup the cluster

Reply:
A bit slow here, so apologies 😄 . By shared function that is private, you mean to not have the shared function exported and then when calling the function for RKE2, just amend the CNI parameter and that's all right?

Reply:
Yes exactly

Reply:
I had a chance to further look into this and we could actually just have the one cluster as is. Then, in the K3S files, we just leave the cni parameter "" when calling it. Locally tested it and this works as expected and we already do this for `cloudCredentialSecretName` in the when calling it in the node driver and custom cluster test files.